gpii/node_modules/contextManager/src/ContextManager.js

```diff
@@ -19,74 +19,136 @@
+        gpii = fluid.registerNamespace("gpii"),
+        $ = fluid.registerNamespace("jQuery");
 
     fluid.registerNamespace("gpii.contextManager");
 
     fluid.defaults("gpii.contextManager", {
         gradeNames: ["fluid.eventedComponent", "autoInit"],
+        members: {
+            currentContext: {},
+            temporalIntervalID: null
+        },
         components: {
             transformer: {
                 type: "gpii.transformer"
+            },
+            lifecycleManager: {
+                type: "gpii.lifecycleManager"
             }
         },
-        handlers: {
-            // contextChanged: {
-            //     route: "/match",
-            //     type: "post"
-            // }, // TODO
-        },
         invokers: {
             evaluateMatch: {
                 funcName: "gpii.contextManager.evaluateMatch",
                 args: [ "{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2" ]
+            },
+            reEvaluateConditions: {
+                funcName: "gpii.contextManager.reEvaluateConditions",
+                args: [ "{that}", "{flowManager}" ]
+            }
+        },
+        listeners: {
+            onCreate: "gpii.contextManager.startTemporalEnvironmentReporter",
+            onDestroy: "gpii.contextManager.stopTemporalEnvironmentReporter"
+        }
+    });
+
+    fluid.defaults("kettle.requests.request.handler.environmentChanged", {
+        gradeNames: ["autoInit", "fluid.littleComponent"],
+        invokers: {
+            handle: {
+                funcName: "gpii.contextManager.environmentChanged",
+                args: ["{flowManager}.contextManager", "{request}.req.body", "{requestProxy}.events.onSuccess"],
+                dynamic: true
             }
         }
     });
 
+    gpii.contextManager.startTemporalEnvironmentReporter = function (that) {
+        that.intervalID = setInterval(function() {
+            gpii.contextManager.environmentChanged(that, {
+                // set in ms since UTC epoch, eg: 1421066816897
+                "http://registry.gpii.net/common/environment/temporal.time": (new Date()).getTime()
+            });
+        }, 20000); //report time every 20 seconds
+    };
+
+    gpii.contextManager.stopTemporalEnvironmentReporter = function (that) {
+        clearInterval(that.intervalID);
+    };
+
+    gpii.contextManager.environmentChanged = function (that, body, event) { // event is optional
+        // add the new context to the existing contexts:
+        var currentContext = that.currentContext;
+        fluid.each(body, function (newValue, contextId) {
+            if (contextId === "timestamp") { // ignore timestamps
+                return;
+            }
+            currentContext[contextId] = newValue;
+        });
+        fluid.log("Context has been updated to: "+JSON.stringify(currentContext, null, 4));
+        // re-evaluate the context and matchData based on the new context
+        that.reEvaluateConditions();
```

Comment:
This should just be expressed as a model listener automatically reacting to the model